ArcadiaTeamsBot/Dialogs/MainDialog.cs

```diff
@@ -4,71 +4,68 @@
     using System.Threading;
     using System.Threading.Tasks;
 
+    using ArcadiaTeamsBot.ServiceDesk.Requests.RequestType;
+    using ArcadiaTeamsBot.ServiceDesk.Requests.RequestTypeFactory;
+
     using MediatR;
 
     using Microsoft.Bot.Builder;
     using Microsoft.Bot.Builder.Dialogs;
     using Microsoft.Bot.Schema;
 
-    public class MainDialog : ComponentDialog
+    internal class MainDialog : ComponentDialog
     {
         private const string NewRequest = "New request";
-        private const string OpenedRequests = "See current requests";
+        private const string OpenedRequests = "Opened requests";
 
-        public MainDialog(IMediator mediator) : base(nameof(MainDialog))
+        public MainDialog(IMediator mediator, IRequestTypeUIFactory requestTypeUIFactory, IEnumerable<RequestTypeUIField> requestTypeUIFields) : base(nameof(MainDialog))
         {
-            this.AddDialog(new RequestsTypeDialog(mediator));
+            this.AddDialog(new RequestsTypeDialog(mediator, requestTypeUIFactory, requestTypeUIFields));
```

Comment:
Could you avoid this and use DI to inject all concrete dialogs into constructor?